tc/core/polyhedral/schedule_transforms.cc

```diff
@@ -823,11 +823,18 @@ isl::multi_union_pw_aff extractDomainToIds(
   for (auto mapping : tree->collect(tree, ScheduleTreeType::MappingFilter)) {
     auto mappingNode = mapping->elemAs<ScheduleTreeElemMappingFilter>();
     auto list = isl::union_pw_aff_list(tree->ctx_, ids.size());
+    bool hasIds = true;
     for (auto id : ids) {
-      CHECK_GT(mappingNode->mapping.count(id), 0) << "no mapping to id " << id;
+      if (mappingNode->mapping.count(id) == 0) {
+        hasIds = false;
+        break;
+      }
       auto idMap = mappingNode->mapping.at(id);
       list = list.add(idMap);
     }
+    if (!hasIds) {
+      continue;
+    }
```

Comment:
Hmm... I wonder if it would be clearer if you would simply collect all the elements with a matching id and then check that the resulting list is either empty (in which case you skip it) or of size equal to `ids.size()`.

Reply:
+1 was going to suggest the same